rust/processor/src/processors/parquet_processors/parquet_events_processor.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    bq_analytics::{
+        create_parquet_handler_loop, generic_parquet_processor::ParquetDataGeneric,
+        ParquetProcessingResult,
+    },
+    db::common::models::events_models::parquet_events::{Event, ParquetEventModel},
+    gap_detectors::ProcessingResult,
+    processors::{parquet_processors::UploadIntervalConfig, ProcessorName, ProcessorTrait},
+    utils::{counters::PROCESSOR_UNKNOWN_TYPE_COUNT, database::ArcDbPool},
+};
+use ahash::AHashMap;
+use anyhow::anyhow;
+use aptos_protos::transaction::v1::{transaction::TxnData, Transaction};
+use async_trait::async_trait;
+use kanal::AsyncSender;
+use serde::{Deserialize, Serialize};
+use std::{fmt::Debug, time::Duration};
+use tracing::warn;
+
+const GOOGLE_APPLICATION_CREDENTIALS: &str = "GOOGLE_APPLICATION_CREDENTIALS";
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ParquetEventsProcessorConfig {
+    pub google_application_credentials: Option<String>,
+    pub bucket_name: String,
+    pub bucket_root: String,
+    pub parquet_handler_response_channel_size: usize,
+    pub max_buffer_size: usize,
+    pub parquet_upload_interval: u64,
+}
+
+impl UploadIntervalConfig for ParquetEventsProcessorConfig {
+    fn parquet_upload_interval_in_secs(&self) -> Duration {
+        Duration::from_secs(self.parquet_upload_interval)
+    }
+}
+
+pub struct ParquetEventsProcessor {
+    connection_pool: ArcDbPool,
+    event_sender: AsyncSender<ParquetDataGeneric<Event>>,
+}
+
+impl ParquetEventsProcessor {
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetEventsProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
+        if let Some(credentials) = config.google_application_credentials.clone() {
```

Comment:
we can create a new Trait to handle this; it seems we're duplicating this logic for every processor.

Reply:
yeah I can extract this out